src/main/java/net/glowstone/entity/objects/GlowPainting.java

```diff
@@ -0,0 +1,334 @@
+    }
+
+    @Override
+    public List<Message> createSpawnMessage() {
+        int x = center.getBlockX();
+        int y = center.getBlockY();
+        int z = center.getBlockZ();
+        String title = getArtTitle();
+
+        return Collections.singletonList(
+            new SpawnPaintingMessage(this.getEntityId(), this.getUniqueId(), title, x, y, z, facing.ordinal())
+        );
+    }
+
+    @Override
+    public Art getArt() {
+        return art;
+    }
+
+    @Override
+    public boolean setArt(Art art) {
+        return this.setArt(art, false);
+    }
+
+    @Override
+    public boolean setArt(Art art, boolean force) {
+        Art oldArt = this.art;
+        setArtInternal(art);
+        updateBoundingBox();
+
+        if (!force && isObstructed()) {
+            setArtInternal(oldArt);
+            updateBoundingBox();
+            return false;
+        }
+
+        respawn();
+
+        return true;
+    }
+
+    private void respawn() {
+        DestroyEntitiesMessage destroyMessage = new DestroyEntitiesMessage(Collections.singletonList(this.getEntityId()));
+        List<Message> spawnMessages = this.createSpawnMessage();
+        Message[] messages = new Message[] {destroyMessage, spawnMessages.get(0)};
+
+        getWorld()
+            .getRawPlayers()
+            .stream()
+            .filter(p -> p.canSeeEntity(this))
+            .forEach(p -> p.getSession().sendAll(messages));
+    }
+
+    public void setArtInternal(Art art) {
+        this.art = art;
+
+        recalculateLocation();
+
+        updateBoundingBox();
+    }
+
+    private void recalculateLocation() {
+        BlockFace rightFace = getRightFace();
+        double modX = rightFace.getModX() * art.getBlockWidth() / 2.0;
+        double modY = art.getBlockHeight() / 2.0;
+        double modZ = rightFace.getModZ() * art.getBlockWidth() / 2.0;
+
+        BlockFace facing = getFacing();
+        if (modX == 0.0) {
+            modX = 0.5 - facing.getModX() * 0.5 - PAINTING_DEPTH / 2;
+        } else if (modZ == 0.0) {
+            modZ = 0.5 - facing.getModZ() * 0.5 - PAINTING_DEPTH / 2;
+        }
+
+        Location add = getTopLeftCorner().add(modX, -modY, modZ);
+        location.setX(add.getX());
+        location.setY(add.getY());
+        location.setZ(add.getZ());
+        location.setPitch(0);
+        location.setYaw(getYaw());
+    }
+
+    @Override
+    public boolean setFacingDirection(BlockFace blockFace, boolean force) {
+        HangingFace oldFace = facing;
+        this.facing = HangingFace.getByBlockFace(blockFace);
+
+        if (!force && isObstructed()) {
+            this.facing = oldFace;
+            return false;
+        }
+
+        respawn();
+
+        return true;
+    }
+
+    @Override
+    public void setFacingDirection(BlockFace blockFace) {
+        setFacingDirection(blockFace, false);
+    }
+
+    @Override
+    public void pulse() {
+        super.pulse();
+
+        if (ticksLived % 11 == 0) {
```

Comment:
Where does the `11` value come from? Do paintings only update every 0.55 seconds?